src/app/gagestats/gagestats.component.ts

```diff
@@ -53,20 +54,25 @@ export class GagestatsComponent implements OnInit {
     this.title = 'Gage Stats';
     this.timestamp = new Date();
     this.showStationType = false;
-    this._nssService.selectedStationType.subscribe((s: Stationtype) => {
-      this.selectedStationType = s;
-      this.selectedAgency = [];
-      console.log(typeof s)
-      if (s) { this.showStationType = true; }
-      window.scrollTo(0, 0);
+    this._nssService.term.subscribe(t => {
+      this.term = t;
+      this._nssService.selectedStationType.subscribe((s: Stationtype) => {
+        this.selectedStationType = s;
+        if (this.selectedStationType > []) {
+          this.showStationType = true;
+        }
+      });
```

Comment:
Usually I try not to nest subscribers inside each other. I'm guessing you did this because you're calling searchStations here. Instead, when you hit `search` in the sidebar it can actually call the `searchStations` function and create a subject for the selected stations, which you can subscribe to here as well. Hope that makes sense!

Also, the station types get updated in the main gagestats component before the search is triggered, so pushing the search button should also update the stationtype subject.

Reply:
I put the `selectedStationType.subscribe` function in with the search button in `gs-sidebar.component.ts` so that the stations are filtered by type when the button is clicked.